tests/foreman/api/test_user.py

```diff
@@ -675,3 +677,255 @@ def test_positive_info_users_ssh_key(self):
             user=self.user, name=ssh_name, key=ssh_key).create()
         self.assertEqual(ssh_name, user_sshkey.name)
         self.assertEqual(ssh_key, user_sshkey.key)
+
+
+@run_in_one_thread
+class ActiveDirectoryUserTestCase(APITestCase):
+    """Implements the LDAP auth User Tests with Active Directory"""
+
+    @classmethod
+    @skip_if_not_set('ldap')
+    def setUpClass(cls):
+        """Fetch necessary properties from settings and Create ldap auth source"""
+        super(ActiveDirectoryUserTestCase, cls).setUpClass()
+        cls.org = entities.Organization().create()
+        cls.loc = entities.Location(organization=[cls.org]).create()
+        cls.sat_url = 'https://{}'.format(settings.server.hostname)
+        cls.ldap_user_name = settings.ldap.username
+        cls.ldap_user_passwd = settings.ldap.password
+        cls.authsource = entities.AuthSourceLDAP(
+            onthefly_register=True,
+            account=cls.ldap_user_name,
+            account_password=cls.ldap_user_passwd,
+            base_dn=settings.ldap.basedn,
+            groups_base=settings.ldap.grpbasedn,
+            attr_firstname=LDAP_ATTR['firstname'],
+            attr_lastname=LDAP_ATTR['surname'],
+            attr_login=LDAP_ATTR['login_ad'],
+            server_type=LDAP_SERVER_TYPE['API']['ad'],
+            attr_mail=LDAP_ATTR['mail'],
+            name=gen_string('alpha'),
+            host=settings.ldap.hostname,
+            tls=False,
+            port='389',
+            location=[cls.loc],
+            organization=[cls.org],
+        ).create()
+
+    def tearDown(self):
+        for user in entities.User().search(query={'search': u'login={}'.format(
```

Comment:
I dont think this is required to run in for a loop. Because it will always return only one user as we are searching with login. And the login will be always unique in entire satellite server. Just the first element in search()[0] is enough.

Reply:
@jyejare That will fail to delete user if user is not present 

```
>               self.ldap_user_name)})[0]
E       IndexError: list index out of range

tests/foreman/api/test_user.py:717: IndexError
```